application/heartbeater.go

```diff
@@ -70,3 +71,9 @@ func (hb *heartbeater) send(tags map[string]string) {
 		log.Printf("heartbeater SendMetric error: %v\n", err)
 	}
 }
+
+func (hb *heartbeater) AddCustomTags(tags map[string]string) {
+	for key, _ := range tags {
+		hb.application.CustomTags[key] = tags[key]
```

Comment:
let's maintain a separate set of maps and send the heartbeat metrics for each map.